app/controllers/spree/api/images_controller_decorator.rb

```diff
@@ -0,0 +1,15 @@
+Spree::Api::ImagesController.class_eval do
+  def update_product_image
+    @product = Spree::Product.find(params[:product_id])
+    authorize! :update, @product
+
+    if @product.images.first.nil?
+      @image = Spree::Image.create(attachment: params[:file], viewable_id: @product.master.id, viewable_type: 'Spree::Variant')
+      respond_with(@image, :status => 201)
+    else
+      @image = @product.master.images.first
```

Comment:
why do we pick the image from the master product here but we go straight to `product.images` in the `if` branch (line 6)? I'm just trying to understand.

Reply:
Looks like we can rid of the call to the `master` association here. According to https://github.com/spree/spree/blob/1-3-stable/core/app/models/spree/product.rb#L63 `images` is delegated to `master`.